docs/source/api-life-cycle.md

```diff
@@ -0,0 +1,186 @@
+4. The API removal may now be made permanent having reached the declared version. Make sure documentation is also cleaned up.
+
+
+In the case that a breaking change has to be made to an existing interface, for example, if a non-optional parameter has to be added to a method, then we treat this as a deprecation case. Steps to take:
+
+
+1. Create a new API that meets the new requirements.
+2. Migrate use cases from the old API to the new API.
+3. Deprecate the old API.
+
+
+## How to Mark?
+
+We want to make the best effort to mark in the most standard way possible, so that it’s easier for users to integrate with their dev tools, like Linters.
+
+<table>
+  <tr>
+   <td><strong>API</strong>
+   </td>
+   <td><strong>Code</strong>
+   </td>
+   <td><strong>Documentation</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Python
+   </td>
+   <td>
+<ul>
+
+<li><em>Deprecated</em>: use <a href="https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.deprecated">typing_extensions.deprecated</a> decorator
+
+<li><em>Experimental</em>: use ExecuTorch's native experimental (TODO not yet implemented) decorator
+</li>
+</ul>
+   </td>
+   <td>
+<ul>
+
+<li>Use <code>.. warning::</code> in docstring of deprecated and experimental APIs, see example usage <a href="https://github.com/pytorch/pytorch/blob/cd8bbdc71a0258292381a7d54c8b353988d02ff4/torch/nn/utils/stateless.py#L170">here</a>.
+</li>
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td>C++
+   </td>
+   <td>
+<ul>
+
+<li>Use the <code>__ET_DEPRECATED</code> and (TODO not yet implemented) <code>__ET_EXPERIMENTAL</code> macro, see example usage <a href="https://github.com/pytorch/executorch/blob/8e0f856ee269b319ac4195509cf31e3f548aa0e8/runtime/executor/program.h#L81">here</a>.
+</li>
+</ul>
+   </td>
+   <td>
+<ul>
+
+<li>Start docstring with “DEPRECATED” and “EXPERIMENTAL”, see example usage <a href="https://github.com/pytorch/executorch/blob/9d859653ae916d0a72f6b2b5c5925bed38832140/runtime/executor/program.h#L139">here</a>.
+</li>
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td>Android
+   </td>
+   <td>TODO
+   </td>
+   <td>TODO
+   </td>
+  </tr>
+  <tr>
+   <td>iOS
+   </td>
+   <td>TODO
+   </td>
+   <td>TODO
+   </td>
+  </tr>
+</table>
+
+
+## Deprecation Period
+
+Here we recommend waiting for at least 2 minor releases before the removal. For example, if a function is marked as _deprecated_ in release 1.3.x, then it can be _deleted_ in 1.5.x or later.
+
+
+## Future Enhancements
+
+
+### Enforcement
+
+We want to implement automated tests to monitor:
+* Were any BC-breaking changes made to stable APIs?
+* Did the removal of an deprecated API happen not sooner than the promised period?
+
+
+### Auto migration
+
+The current deprecation policy still requires users to take actions to migrate off from deprecated APIs. Ideally, we should automate this process. We want to leverage tools like [TorchFix](https://pypi.org/project/TorchFix/) to auto-migrate uses of deprecated APIs when possible.
```

Comment:
Please remove this section. We do want to work on it, but it's not relevant to the policy itself.
```suggestion
```